### PR TITLE
Sanitize HTML for table and publication descriptions

### PR DESCRIPTION
**Pull request by alisonrclarke**

### CODE REVIEW

File: hepdata/config.py [hunk 1]
```diff
@@ -300,6 +300,11 @@ def _(x):
 
 LOGGING_SENTRY_CELERY = True  # for invenio-logging
 
+# HTML attributes and tags allowed in record descriptions
+# If these are modified, update tips.rst in hepdata-submission
+ALLOWED_HTML_ATTRS = {'*': ['class'], 'a': ['href', 'title', 'name', 'class', 'rel'], 'abbr': ['title'], 'acronym': ['title']}
+ALLOWED_HTML_TAGS = ['a', 'abbr', 'acronym', 'b', 'blockquote', 'br', 'code', 'div', 'em', 'i', 'li', 'ol', 'p', 'pre', 'span', 'strike', 'strong', 'sub', 'sup', 'u', 'ul']
```
Comment (GraemeWatt) [inline hepdata/config.py:306]: I know these attributes and tags were copied from Invenio-Formatter, but do you see a use case for the `class` attribute?  I suppose an Uploader could use a class name given in a HEPData style sheet to achieve the same styling, but then we'd need to document possible class names for this to be useful.  I'm wondering whether it would be simpler to remove `class` as a possible attribute and update `tips.rst` in the hepdata-submission docs.

File: hepdata/modules/records/templates/hepdata_records/components/table_list.html [hunk 2]
```diff
@@ -124,7 +124,7 @@ <h4 itemprop="name">{{ table.name }}</h4>
                     {% endif %}
 
                     {% if table.description != '.' %}
-                        <p itemprop="description">{{ table.description }}</p>
+                        <p itemprop="description">{{ table.description|safe }}</p>
```
Comment (GraemeWatt) [inline hepdata/modules/records/templates/hepdata_records/components/table_list.html:127]: Maybe it'd be better to escape HTML in the list of tables (i.e. remove `|safe`) as was the previous behaviour.  The list of tables is mainly used by the user to select the table of interest.  If the text contains an HTML link, maybe the user would click the link accidentally instead of selecting the table.  The use of `sanitize_html` can then be removed from `process_data_tables` above.

File: hepdata/utils/miscellaneous.py [hunk 3]
```diff
@@ -20,9 +20,52 @@
...
+        False (escapes rather than strips disallowed tags).
+
+    Use this function when you need to include unescaped HTML that contains
+    user provided data.
+    """
+    from flask import current_app
+
+    # The lines below were copied from invenio-formatter but with the addition
+    # of the strip parameter.
+    value = value.strip()
```
Comment (GraemeWatt) [inline hepdata/utils/miscellaneous.py:53]: The use of `strip` gives an exception if `value` is `None`, for example, for a newly created record without a data abstract.  I guess you just need to add something like:
```python
if value is None:
    return value
```

File: tests/miscellaneous_test.py [hunk 4]
```diff
@@ -62,3 +62,43 @@ def test_file_extractor(app):
             file = get_file_in_directory(extract_dir, 'yaml')
             assert (file is not None)
 
+
+def test_sanitize_html(app):
+    with app.app_context():
+        test_cases = [
+            ("<b>Here is some bold text</b> and <script>here is a dodgy script</script>",
```
Comment (GraemeWatt) [inline tests/miscellaneous_test.py:69]: Please add the test case `None` where the sanitized result is also `None`.

**Review by GraemeWatt — CHANGES_REQUESTED** [review]
Thanks, this looks good.  Please address the minor comments before I merge.

File: hepdata/modules/records/templates/hepdata_records/components/table_list.html (hunk 2 above)
Comment (alisonrclarke) [reply hepdata/modules/records/templates/hepdata_records/components/table_list.html:127]: I see what you mean but I suggest we use `sanitize_html` with no tags allowed and `strip=True` so that we just remove any HTML elements and leave the plain text, rather than have the table list littered with `&lt;` etc.

**Review by GraemeWatt — APPROVED** [review]
